app/models/miq_alert_status.rb

```diff
@@ -1,4 +1,19 @@
+require 'ancestry'
+
 class MiqAlertStatus < ApplicationRecord
+  SEVERITY_LEVELS = %w(error warning info).freeze
+
+  has_ancestry
   belongs_to :miq_alert
   belongs_to :resource, :polymorphic => true
+  has_many :miq_alert_status_actions, -> { order "created_at" }, :dependent => :destroy
+  virtual_column :assignee, :type => :string
+
+  def assignee
+    miq_alert_status_actions.where(:action_type => %w(assign unassign)).last.try(:assignee)
```

Comment:
Perhaps encapsulate the `:action_type => %w(assign unassign)`, or even more of this logic, in a MiqAlertStatusAction scope.

Ordered association: neat, though I've seen a recommendation to avoid implicit ordering, or capture it in separate scope: http://blog.carbonfive.com/2016/11/16/rails-database-best-practices/.

LGTM either way.